perun-core/src/main/java/cz/metacentrum/perun/core/impl/GroupsManagerImpl.java

```diff
@@ -977,6 +994,40 @@ public MemberGroupStatus getTotalMemberGroupStatus(PerunSession session, Member
 		}
 	}
 
+	@Override
+	public Map<Member, MemberGroupStatus> getTotalGroupStatusForMembers(PerunSession session, Group group, List<Member> members) {
+		List<Integer> memberIds = new ArrayList<>();
+		members.forEach(member -> memberIds.add(member.getId()));
+
+		MapSqlParameterSource parameters = new MapSqlParameterSource();
+		parameters.addValue("groupId", group.getId());
+		parameters.addValue("memberIds", memberIds);
+
+		try {
+			Map<Member, List<Integer>> map = namedParameterJdbcTemplate.query("select " + MembersManagerImpl.memberMappingSelectQuery + ", source_group_status FROM groups_members" +
+				" join members on groups_members.member_id=members.id where group_id=(:groupId) and member_id in (:memberIds)", parameters, MEMBER_MEMBERGROUPSTATUS_EXTRACTOR);
+
+			if (map == null) {
+				return null;
+			}
```

Comment:
I think the result can never be `null` since you return an empty map from the results extractor.

Reply:
I think so, but since the method `query` is annotated as `nullable`, the check from idea does not like it... I guess I could assert it's not null or something like that?

Reply:
I see. If that is the case, you can keep it that way.